Zend/tests/offsets/test_offset_helpers.inc

```diff
@@ -0,0 +1,171 @@
+        return "STDERR";
+    }
+    if ($container instanceof stdClass) {
+        return "new stdClass()";
+    }
+    if ($container instanceof B) {
+        return "new B()";
+    }
+    if ($container instanceof A) {
+        return "new A()";
+    }
+    if ($container instanceof ArrayObject) {
+        return "new ArrayObject()";
+    }
+    return var_export($container, true);
+}
+function makeOffset($offset) {
+    if ($offset === PHP_INT_MIN) {
+        return "PHP_INT_MIN";
+    }
+    if ($offset === PHP_INT_MAX) {
+        return "PHP_INT_MAX";
+    }
+    return var_export($offset, true);
+}
+
+/* Taken from run-tests.php */
+function expectf_to_regex(string $wanted): string
+{
+    $wanted_re = preg_quote($wanted);
+    return strtr($wanted_re, [
+        '%e' => preg_quote(DIRECTORY_SEPARATOR, '/'),
+        '%s' => '[^\r\n]+',
+        '%S' => '[^\r\n]*',
+        '%a' => '.+',
+        '%A' => '.*',
+        '%w' => '\s*',
+        '%i' => '[+-]?\d+',
+        '%d' => '\d+',
+        '%x' => '[0-9a-fA-F]+',
+        '%f' => '[+-]?(?:\d+|(?=\.\d))(?:\.\d+)?(?:[Ee][+-]?\d+)?',
+        '%F' => '([+-]?(?:\d+|(?=\.\d))(?:\.\d+)?(?:[Ee][+-]?\d+)?)|(NAN)|(INF)',
+        '%c' => '.',
+        '%0' => '\x00',
+    ]);
+}
+
+class A implements ArrayAccess {
+    public function offsetSet($offset, $value): void {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        var_dump($value);
+    }
+    public function offsetGet($offset): mixed {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        return "dummy";
+    }
+    public function offsetUnset($offset): void {
+        var_dump(__METHOD__);
+        var_dump($offset);
+    }
+    public function offsetExists($offset): bool {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        return false;
+    }
+}
+
+class B extends ArrayObject {
+    public function offsetSet($offset, $value): void {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        var_dump($value);
+    }
+    public function offsetGet($offset): mixed {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        return "dummy";
+    }
+    public function offsetUnset($offset): void {
+        var_dump(__METHOD__);
+        var_dump($offset);
+    }
+    public function offsetExists($offset): bool {
+        var_dump(__METHOD__);
+        var_dump($offset);
+        return false;
+    }
+    public function append(mixed $value) : void{
+        var_dump(__METHOD__);
+        var_dump($value);
+    }
+}
+
+$containers = [
+    null,
+    false,
+    true,
+    4,
+    5.5,
+    '10',
+    '25.5',
+    'string',
+    [],
+    STDERR,
+    new stdClass(),
+    new ArrayObject(),
+    new A(),
+    new B(),
+];
+
+$offsets = [
+    null,
+    false,
+    true,
+    4,
+    5.5,
+    6.0,
+    -12,
+    -13.5,
+    -14.0,
+    //PHP_INT_MAX,
+    //PHP_INT_MIN,
+    PHP_INT_MAX * 2,
+    PHP_INT_MIN * 2,
+    INF,
+    -INF,
+    NAN,
```

Comment:
TODO: Add ``STDERR`` as an offset for a resource offset type test.

And also array and stdClass

Reply:
Done